clicommand/cache_restore.go

```diff
@@ -0,0 +1,99 @@
+package clicommand
+
+import (
+	"context"
+	"slices"
+
+	"github.com/buildkite/agent/v3/internal/cache"
+	"github.com/urfave/cli"
+)
+
+const cacheRestoreHelpDescription = `Usage:
+
+    buildkite-agent cache restore [options]
+
+Description:
+
+Restores files from the cache for the current job based on the cache configuration
+defined in your cache config file (defaults to .buildkite/cache.yml).
+
+The cache configuration file defines which files or directories should be restored
+and their associated cache keys. Caches are scoped by organization, pipeline, and
+branch. If an exact cache match is not found, the command will attempt to use
+fallback keys if defined in your cache configuration.
+
+Note: This feature is currently in development and subject to change. It is not
+yet available to all customers.
```

Comment:
might be good to put an `EXPERIMENTAL` callout in here, as that's usually the verbiage we use to denote something that might change